homeassistant/components/xiaomi_miio/humidifier.py

```diff
@@ -197,66 +171,61 @@ def __init__(self, name, device, entry, unique_id, coordinator):
         """Initialize the plug switch."""
         super().__init__(name, device, entry, unique_id, coordinator)
         if self._model in [MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1]:
-            self._available_modes = AVAILABLE_MODES_CA1_CB1
-            self._min_humidity = 30
-            self._max_humidity = 80
+            self._attr_available_modes = AVAILABLE_MODES_CA1_CB1
+            self._attr_min_humidity = 30
+            self._attr_max_humidity = 80
             self._humidity_steps = 10
         elif self._model in [MODEL_AIRHUMIDIFIER_CA4]:
-            self._available_modes = AVAILABLE_MODES_CA4
-            self._min_humidity = 30
-            self._max_humidity = 80
+            self._attr_available_modes = AVAILABLE_MODES_CA4
+            self._attr_min_humidity = 30
+            self._attr_max_humidity = 80
             self._humidity_steps = 100
         elif self._model in MODELS_HUMIDIFIER_MJJSQ:
-            self._available_modes = AVAILABLE_MODES_MJJSQ
-            self._min_humidity = 30
-            self._max_humidity = 80
+            self._attr_available_modes = AVAILABLE_MODES_MJJSQ
+            self._attr_min_humidity = 30
+            self._attr_max_humidity = 80
             self._humidity_steps = 10
         else:
-            self._available_modes = AVAILABLE_MODES_OTHER
-            self._min_humidity = 30
-            self._max_humidity = 80
+            self._attr_available_modes = AVAILABLE_MODES_OTHER
+            self._attr_min_humidity = 30
+            self._attr_max_humidity = 80
             self._humidity_steps = 10
 
-        self._state = self.coordinator.data.is_on
+        self._attr_is_on = self.coordinator.data.is_on
         self._attributes.update(
             {
                 key: self._extract_value_from_attribute(self.coordinator.data, value)
                 for key, value in AVAILABLE_ATTRIBUTES.items()
             }
         )
         self._target_humidity = self._attributes[ATTR_TARGET_HUMIDITY]
-        self._mode = self._attributes[ATTR_MODE]
-
-    @property
-    def is_on(self):
-        """Return true if device is on."""
-        return self._state
+        self._attr_mode = self._attributes[ATTR_MODE]
 
     @callback
     def _handle_coordinator_update(self):
         """Fetch state from the device."""
-        self._state = self.coordinator.data.is_on
+        self._attr_is_on = self.coordinator.data.is_on
         self._attributes.update(
             {
                 key: self._extract_value_from_attribute(self.coordinator.data, value)
                 for key, value in AVAILABLE_ATTRIBUTES.items()
             }
         )
         self._target_humidity = self._attributes[ATTR_TARGET_HUMIDITY]
-        self._mode = self._attributes[ATTR_MODE]
+        self._attr_mode = self._attributes[ATTR_MODE]
```

Comment:
Do not assign the mode attribute directly here. We have device depended code for this.